docs/core/diagnostics/observability-with-otel.md

```diff
@@ -0,0 +1,429 @@
+---
+title: .NET Observability with OpenTelemetry
+description: An introduction to observing .NET apps with OpenTelemetry
+ms.date: 6/14/2023
+ms.topic: conceptual
+---
+
+# .NET observability with OpenTelemetry
+
+When you run a distributed application, you want to know how well the app is performing and to detect potential problems before they become larger. One way that you can do this is by monitoring specific instrumentation emitted by the application and looking for changes in metrics and logs.
+
+## What is observability
+
+Observability in the context of a distributed system is the ability to monitor and analyze telemetry about the state of each component, to be able to observe changes in performance, and to diagnose why those changes occur. Unlike debugging, which is invasive and can affect the operation of the application, observability is intended to be transparent to the primary operation and have a small enough performance impact that it can be used continuously.
+
+Observability is commonly done using a combination of:
+
+- Logs - record individual operations, such as an incoming request, a failure in a specific component, or an order being placed.
+- Metrics - measuring counters and gauges such as number of completed requests, active requests, widgets that have been sold; or a histogram of the request latency.
+- Distributed tracing - track requests and activities across components in a distributed system so that you can see where time is spent and track down specific failures.
+
+Together, logs, metrics, and distributed tracing are known as the *3 pillars of observability*.
+
+Each pillar might include telemetry data from:
+
+- The .NET runtime, such as the garbage collector or JIT.
+- Libraries, such as from Kestrel (the ASP.NET web server) and `HttpClient`.
+- Application-specific telemetry that's fired by your code.
+
+## Observability approaches in .NET
+
+There are a few different ways to achieve observability in .NET applications:
+
+- Explicitly in code, by referencing and using an SDK such as OpenTelemetry. If you have access to the source code and can rebuild the app, then this is the most powerful and configurable mechanism.
+- Out-of-process using [EventPipe](./eventpipe.md). Tools such as [dotnet-monitor](./dotnet-monitor.md) can listen to logs and metrics and then process them without affecting any code.
+- [Using a startup hook](https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md), assemblies can be injected into the process that can then collect instrumentation. This mechanism enables creating instrumentation for libraries that weren't designed for it but can be fragile to version conflicts between application and instrumentation dependencies. An example of this approach is [OpenTelemetry .NET Automatic Instrumentation](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/README.md).
+
+## What is OpenTelemetry
```

Comment:
I started but I'm not sure if that just adds more confusion, as it leads to providing more details. -->

For example, an app's code makes the following logging call:

``` c#
logger.LogInformation("Customer {customerid} added {n} of {productid} to shopping cart {cartid}", customerId, itemCount, productId, cartId);
```

to provide status about what is happening during a shopping experience. The message is collected, either via the ILogger Console Provider and `stdout`, or via another ILogger provider, such as one that does semantic logging. The message and parameters need to be egressed out of the process to a logging data store. In a distributed system, it is commonly a separate service, that is used to collect data from multiple services, and potentially multiple instances of each of those services, so needs to be sent across the network. Rather than sending each log message as it happens, they are usually buffered and asynchronously sent as a batch. Logging information not only includes the message, but also a timestamp, and information on the process that fired it. If [`Scopes`](../extensions/logging#log-scopes) are used it can include addition information such as the TraceID and SpanId for distributed tracing, so that log messages can be correlated with tracing activity. 

The log store will store all the log records together, and have query capabilities that can be used to analyze what happened. For example, being able to query all the activity for a specific shopping cart or user. By counting the results charts can be created to report on metrics that may not explicitly collected by the application - this is useful for drilling down when investigating issues to see when a problem may have occurred.  

Logging is usually asynchronous and best-effort - so should not be used as the basis for how billing or order processing is built - it should be treated as ancillary information about how the service has operated, not be used for the operation of the service.

Reply:
Perhaps there is a middle ground where the description is specific to OpenTelemetry, but more detailed than the list you have below? My worry is that I am anticipating devs that read the list below won't really understand it and won't click through each link to gain that understanding. Perhaps something like:

[OpenTelemetry](https://opentelemetry.io/) (OTel) is a cross-platform, open standard for collecting and emitting telemetry data. OpenTelemetry includes:
1. [APIs](https://opentelemetry.io/docs/concepts/instrumentation/manual/) used by app developers and library authors to record telemetry data as code is running.
2. [APIs](some_link) that app developers use to configure what portion of the recorded data will be sent across the network, where it will be sent to, and how it may be filtered, buffered, enriched, and transformed.
3. [Semantic conventions](https://github.com/open-telemetry/semantic-conventions) provide guidance on naming and content of telemetry data. It is important for the apps that produce telemetry data and the tools that receive the data to agree on what
different kinds of data means and what sorts of data are useful so that the tools can provide effective analysis.
4. An interface for [exporters](https://opentelemetry.io/docs/concepts/components/#exporters). Exporters are plugins that allow telemetry data to be transmitted in specific formats to different telemetry backends.
5. [OTLP](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/README.md) is a vendor neutral network protocol option for transmitting telemetry data. Some tools and vendors support this protocol in addition to pre-existing proprietary protocols they may have.